server/player.lua

```diff
@@ -420,11 +420,16 @@ end
 ---On player logout
 ---@param source Source
 function Logout(source)
+    local player = GetPlayer(source)
+    if not player then return end
+    local playerState = Player(source)?.state
```

Comment:
If the player drops, won't the state bag be destroyed?

Reply:
It's not destroyed, at least in the playerDropped event we still have access to the statebag.